<a href="https://colab.research.google.com/github/eospgonz10/statistical_analytics_homicide/blob/main/sesiones_practicas/sp_1_juan_jose_garcia_alvarez_estiven_ospina_gonzalez.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <span style="color:Blue;"><strong>1. Introducción</strong></span>

<div style="background-color:#E6F4FF; padding:10px; border-radius:8px;">
<b>1. Breve presentación del problema o pregunta de investigación.</b>  
</div>

¿Qué factores sociales, demográficos y geográficos influyen en las tendencias de homicidios en Estados Unidos?

<br> 
<div style="background-color:#E8F5E9; padding:10px; border-radius:8px;">
<b> 2. Contexto: ¿por qué es relevante este problema?</b>  
</div>

Los **homicidios** constituyen una de las principales causas de mortalidad prevenible en Estados Unidos, con profundas implicaciones sociales, económicas y de salud pública. Más allá de las cifras, este fenómeno puede refleja desigualdades estructurales y vulnerabilidades en segmentos específicos de la población.

Comprender qué factores influyen en las tendencias de homicidios es relevante por varias razones:

- **Vulnerabilidad geográfica**: Existen zonas marginadas y con menor acceso a servicios de seguridad y bienestar donde la incidencia de homicidios es significativamente mayor.
- **Disparidades demográficas**: Ciertos grupos étnicos y por sexo pueden presentarse víctimas en proporciones desequilibradas, lo que requiere análisis diferenciado.
- **Factores etarios**: Las edades vulnerables varían según contextos geográficos y sociales, indicando diferentes dinámicas de violencia.
- **Patrones espaciales y temporales**: Identificar patrones permite anticipar y prevenir concentraciones de violencia en zonas específicas.

La literatura ha reconocido que factores como la **pobreza**, el **desempleo**, la **segregación racial**, el **acceso a armas**, la **estructura demográfica** y la **densidad poblacional** inciden en las tasas de homicidios, pero su interacción y peso relativo aún merece análisis actualizados.

<div style="background-color:#FFF9C4; padding:10px; border-radius:8px;">
<b> 3. Relación entre el problema y la base de datos seleccionada. </b>  
</div>

Para abordar esta pregunta de investigación, utilizamos una base de datos compilada de registros de homicidios en Estados Unidos, disponible en la plataforma **[Kaggle](https://www.kaggle.com/code/harshilahalpara/homicide)**. Este dataset recopila información detallada de casos de homicidios, incluyendo:

- **Variables geográficas**: Estado, ciudad y otras ubicaciones que permiten identificar zonas de concentración.
- **Variables demográficas**: Género y etnia de las víctimas.
- **Variables temporales**: Año, mes y día del evento, para identificar patrones estacionales y tendencias.
- **Características contextuales**: Información sobre el tipo de arma, circunstancias del evento y otras variables que proporcionan contexto sobre la naturaleza de los homicidios.

Esta información nos permite no solo cuantificar las tendencias globales de homicidios, sino también desagrupar el análisis por regiones, grupos demográficos y períodos temporales, facilitando la identificación de vulnerabilidades específicas.

In [3]:
# Librerías
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

# Estilo general
sns.set_theme(style="whitegrid")
plt.figure(figsize=(12, 6))

# Crear dataset
df = pd.read_csv("../data/database.csv")

C:\Users\estiv\AppData\Local\Temp\ipykernel_21308\3261680054.py:11: DtypeWarning: Columns (16) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../data/database.csv")


<Figure size 1200x600 with 0 Axes>

#Descripción de la base de datos

In [8]:
df.head()

Record ID Agency Code Agency Name       Agency Type       City   State  \
0          1     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
1          2     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
2          3     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
3          4     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   
4          5     AK00101   Anchorage  Municipal Police  Anchorage  Alaska   

   Year    Month  Incident              Crime Type  ... Victim Ethnicity  \
0  1980  January         1  Murder or Manslaughter  ...          Unknown   
1  1980    March         1  Murder or Manslaughter  ...          Unknown   
2  1980    March         2  Murder or Manslaughter  ...          Unknown   
3  1980    April         1  Murder or Manslaughter  ...          Unknown   
4  1980    April         2  Murder or Manslaughter  ...          Unknown   

  Perpetrator Sex  Perpetrator Age               Perpetrator Race  \
0            Male               15  Native American/Alaska Native   
1            Male               42                          White   
2         Unknown                0                        Unknown   
3            Male               42                          White   
4         Unknown                0                        Unknown   

  Perpetrator Ethnicity  Relationship         Weapon Victim Count  \
0               Unknown  Acquaintance   Blunt Object            0   
1               Unknown  Acquaintance  Strangulation            0   
2               Unknown       Unknown        Unknown            0   
3               Unknown  Acquaintance  Strangulation            0   
4               Unknown       Unknown        Unknown            0   

  Perpetrator Count Record Source  
0                 0           FBI  
1                 0           FBI  
2                 0           FBI  
3                 0           FBI  
4                 1           FBI  

[5 rows x 24 columns]

In [7]:
# Número de registros y variables
print(f"Número de registros: {df.shape[0]}")
print(f"Número de variables: {df.shape[1]}")

Número de registros: 638454
Número de variables: 24


In [ ]:
# Clasificación de variables
categoricas = df.select_dtypes(include=['object']).columns.tolist()
numericas = df.select_dtypes(include=['int64', 'float64']).columns.tolist()

print("🔹 Variables categóricas:")
print("\n".join([f"- {col}" for col in categoricas]))

print("\n🔸 Variables numéricas:")
print("\n".join([f"- {col}" for col in numericas]))

🔹 Variables categóricas:
- Agency Code
- Agency Name
- Agency Type
- City
- State
- Month
- Crime Type
- Crime Solved
- Victim Sex
- Victim Race
- Victim Ethnicity
- Perpetrator Sex
- Perpetrator Age
- Perpetrator Race
- Perpetrator Ethnicity
- Relationship
- Weapon
- Record Source

🔸 Variables numéricas:
- Record ID
- Year
- Incident
- Victim Age
- Victim Count
- Perpetrator Count


In [ ]:
# Descripción general
print("\nDescripción de variables numéricas:")
print(df.describe())


Descripción de variables numéricas:
          Record ID           Year       Incident     Victim Age  \
count  638454.00000  638454.000000  638454.000000  638454.000000   
mean   319227.50000    1995.801102      22.967924      35.033512   
std    184305.93872       9.927693      92.149821      41.628306   
min         1.00000    1980.000000       0.000000       0.000000   
25%    159614.25000    1987.000000       1.000000      22.000000   
50%    319227.50000    1995.000000       2.000000      30.000000   
75%    478840.75000    2004.000000      10.000000      42.000000   
max    638454.00000    2014.000000     999.000000     998.000000   

        Victim Count  Perpetrator Count  
count  638454.000000      638454.000000  
mean        0.123334           0.185224  
std         0.537733           0.585496  
min         0.000000           0.000000  
25%         0.000000           0.000000  
50%         0.000000           0.000000  
75%         0.000000           0.000000  
max        10.

#Análisis inicial de variables

In [ ]:
# Conteo de valores faltantes ordenado de mayor a menor
faltantes = df.isnull().sum().sort_values(ascending=False)

print("\nConteo de valores faltantes por variable:")
print(faltantes[faltantes > 0])  # Solo muestra variables con al menos un valor faltante


Conteo de valores faltantes por variable:
Series([], dtype: int64)


In [ ]:
# Conteo de valores únicos por variable
print("\nConteo de valores únicos por variable:")
print(df.nunique().sort_values(ascending=False))


Conteo de valores únicos por variable:
Record ID                638454
Agency Code               12003
Agency Name                9216
City                       1782
Incident                   1000
Perpetrator Age             191
Victim Age                  101
State                        51
Year                         35
Relationship                 28
Weapon                       16
Month                        12
Perpetrator Count            11
Victim Count                 11
Agency Type                   7
Victim Race                   5
Perpetrator Race              5
Victim Sex                    3
Perpetrator Sex               3
Victim Ethnicity              3
Perpetrator Ethnicity         3
Crime Solved                  2
Crime Type                    2
Record Source                 2
dtype: int64


In [ ]:
# Variables más importantes
variables_clave = ['City', 'Incident', 'State', 'Year', 'Relationship', 'Weapon', 'Crime Solved', 'Crime Type', 'Perpetrator Age', 'Victim Age',  'Perpetrator Count', 'Victim Count', 'Perpetrator Race', 'Victim Race', 'Perpetrator Sex', 'Victim Sex']
print("\nVista rápida de variables clave:")
print(df[variables_clave].head())
# Vista rápida de las variables clave
print("\nEstadísticas descriptivas de variables clave:")
print(df[variables_clave].describe())


Vista rápida de variables clave:
        City  Incident   State  Year  Relationship         Weapon  \
0  Anchorage         1  Alaska  1980  Acquaintance   Blunt Object   
1  Anchorage         1  Alaska  1980  Acquaintance  Strangulation   
2  Anchorage         2  Alaska  1980       Unknown        Unknown   
3  Anchorage         1  Alaska  1980  Acquaintance  Strangulation   
4  Anchorage         2  Alaska  1980       Unknown        Unknown   

  Crime Solved              Crime Type Perpetrator Age  Victim Age  \
0          Yes  Murder or Manslaughter              15          14   
1          Yes  Murder or Manslaughter              42          43   
2           No  Murder or Manslaughter               0          30   
3          Yes  Murder or Manslaughter              42          43   
4           No  Murder or Manslaughter               0          30   

   Perpetrator Count  Victim Count               Perpetrator Race  \
0                  0             0  Native American/Alaska Na

#Visualizaciones exploratorias

## <span style="color:Blue;"><strong>5. Interpretaciones iniciales</strong></span>

<div style="background-color:#E6F4FF; padding:10px; border-radius:8px;">
<b>¿Qué tendencias o patrones básicos se observan?</b>
</div>

Basado en el análisis exploratorio del dataset de homicidios en Estados Unidos, se identifican los siguientes patrones fundamentales:

- **Concentración geográfica**: California y Texas concentran aproximadamente el 26% del total de homicidios (16% y 10% respectivamente), sugiriendo que la violencia homicida está altamente localizada en determinadas regiones.

- **Vulnerabilidad etaria definida**: La mayor parte de las víctimas registran edades entre 19 y 39 años, configurando un rango etario identificable. 

- **Patrón de género en victimización**: El 77% de las víctimas son hombres, mientras que el 63% de los victimarios identificados también son hombres.

- **Asimetría en la identificación de victimarios**: No existe una inclinación clara respecto a factores específicos de los victimarios más allá del género masculino predominante. Esta limitación en la caracterización del perfil del agresor dificulta la formulación de hipótesis sobre motivaciones, contextos sociales o factores de riesgo del perpetrador.

---

<div style="background-color:#E8F5E9; padding:10px; border-radius:8px;">
<b>¿Qué limitaciones tiene la base de datos?</b>
</div>

La base de datos presenta varias limitaciones metodológicas importantes que deben considerarse en análisis posteriores:

1. **Sesgos de cobertura geográfica**: La sobrerrepresentación de estados como California y Texas puede deberse a diferencias en sistemas de reporte, disponibilidad de datos históricos o variaciones en protocolos de registro entre jurisdicciones. Esto limita la generalización de hallazgos a nivel nacional.

2. **Información incompleta sobre victimarios**: El análisis revela que hay una caracterización limitada del perfil del victimario más allá del género. Faltan variables contextuales sobre motivación, antecedentes criminales o relación con la víctima.

3. **Sesgo potencial en casos resueltos**: La variable "Crime Solved" puede estar correlacionada con características demográficas de víctimas (posible sesgo)

4. **Falta de contexto socioeconómico**: La base de datos no incluye indicadores de pobreza, desempleo, densidad poblacional, acceso a servicios o variables socioeconómicas que podrían explicar la concentración geográfica observada.

---

<div style="background-color:#FFF9C4; padding:10px; border-radius:8px;">
<b>¿Qué hipótesis iniciales podrían plantearse para análisis posteriores?</b>
</div>

Con base en los patrones observados, se proponen las siguientes hipótesis para investigaciones futuras:

**Hipótesis geográficas:**
- La concentración de homicidios en California y Texas está asociada a factores estructurales (densidad poblacional, desigualdad económica) más que a factores coyunturales, lo que requeriría análisis comparativos entre estados.

**Hipótesis demográficas:**
- El patrón intra-género (hombres victimizando hombres en 63% de casos) está vinculado a dinámicas de violencia relacionada con conflictos de poder, territorios u organizaciones específicas, diferenciándose de otras formas de violencia interpersonal.
- El rango etario 19-39 años representa la población económicamente activa y potencialmente involucrada en dinámicas de conflicto (laborales, sociales, criminales) que elevan el riesgo de victimización.
- Las características demográficas de víctimas y perpetradores pueden estar correlacionadas (proximidad geográfica, grupo racial/étnico similar), sugiriendo homicidios comunitarios o criminalidad localizada.